plugin/vim-arsync.vim

```diff
@@ -73,6 +73,10 @@ function! ARsync(direction)
             endif
         endif
 
+        if has_key(l:conf_dict, "sleep_before_sync")
+            exe 'sleep '.l:conf_dict['sleep_before_sync']
```

Comment:
I'm not a huge fan of this, this basically blocks vim for x seconds, it completely defeats the purpose of the plugin which is to run asynchronously in the background, isnt there another solution to wait in a background job without blocking vim ?

Reply:
i fixed that! but i was not sure if really to push it or not since i couldn't yet make it to take a parameter, it looks like this:

```
function! AutoSync()
    let l:conf_dict = LoadConf()
    if has_key(l:conf_dict, 'auto_sync_up')
        if l:conf_dict['auto_sync_up'] == 1
            if has_key(l:conf_dict, 'sleep_before_sync')
                autocmd BufWritePost,FileWritePost * call timer_start(3000, { -> execute("call ARsync('up')", "")})
            else
                autocmd BufWritePost,FileWritePost * ARsyncUp
            endif
            " echo 'Setting up auto sync to remote'
        endif
    endif
endfunction
```

Here it will wait 3000 miliseconds before executing the sync, without blocking the vim.

Reply:
This works with the variable:

```vimscript
function! AutoSync()
    let l:conf_dict = LoadConf()
    if has_key(l:conf_dict, 'auto_sync_up')
        if l:conf_dict['auto_sync_up'] == 1
            if has_key(l:conf_dict, 'sleep_before_sync')
                let g:sleep_time = l:conf_dict['sleep_before_sync']*1000
                autocmd BufWritePost,FileWritePost * call timer_start(g:sleep_time, { -> execute("call ARsync('up')", "")})
            else
                autocmd BufWritePost,FileWritePost * ARsyncUp
            endif
            " echo 'Setting up auto sync to remote'
        endif
    endif
endfunction
```

Reply:
For this particular one, could you make the change (remove the other lines of exec and put my working code) and then commit and push to your repo, it should automatically add the new commits to this pull request

Reply:
ohh, okey, but in the evening.

Reply:
ok no problem :)

Reply:
Hello there, i made the change here https://github.com/KenN7/vim-arsync/pull/13/commits/b37e81e2a14bb3ea6cc4da4e9022d45977456395

But there is a catch,  g:sleep_time will only initialize once, when entering the vim, so, if you, during development, decide that the delay is too big and want to change it in .vim-arsync, you'll have to re-enter vim for the value to take effect

Reply:
With my latest addition it also updates if you change dir, for now i think its fine like this.